### PR TITLE
feat: implement swap quote received event

### PR DESCRIPTION
**Pull request by lynnshaoyu**
Screen Shot 2022-07-21 at 6 47 53 PM

also this issue: https://uniswaplabs.atlassian.net/browse/WEB-623?atlOrigin=eyJpIjoiODdlZTM3MDY2MGM5NDZhMDg4ZmVkZjhlYjI1NmY3YTYiLCJwIjoiaiJ9

### CODE REVIEW

File: src/components/swap/SwapDetailsDropdown.tsx [hunk 1]
```diff
@@ -11,15 +12,19 @@ import Row, { RowBetween, RowFixed } from 'components/Row'
...
 import styled, { keyframes, useTheme } from 'styled-components/macro'
 import { HideSmall, ThemedText } from 'theme'
+import { computeRealizedLPFeePercent } from 'utils/prices'
 
 import { AdvancedSwapDetails } from './AdvancedSwapDetails'
+import { getPriceImpactPercent } from './AdvancedSwapDetails'
 import GasEstimateBadge from './GasEstimateBadge'
 import { ResponsiveTooltipContainer } from './styleds'
+import { formatPercentInBasisPointsNumber } from './SwapModalFooter'
+import { getNumberFormattedToDecimalPlace } from './SwapModalFooter'
```
Comment (cmcewen) [inline src/components/swap/SwapDetailsDropdown.tsx:27]: could you move these two functions to a utils file? maybe in the analytics folder? also you should be able to import in one line - VSCode has an "organize imports" setting which makes it nicer

File: src/components/swap/SwapDetailsDropdown.tsx [hunk 2]
```diff
@@ -131,6 +159,12 @@ export default function SwapDetailsDropdown({
   const theme = useTheme()
   const { chainId } = useWeb3React()
   const [showDetails, setShowDetails] = useState(false)
+  const [isFirstPriceFetch, setIsFirstPriceFetch] = useState(true)
+  const lpFeePercent = trade ? computeRealizedLPFeePercent(trade) : undefined
```
Comment (cmcewen) [inline src/components/swap/SwapDetailsDropdown.tsx:163]: if this is computed from trade, can't this be computed in `formatAnalyticsEventProperties`?

File: src/components/swap/SwapDetailsDropdown.tsx [hunk 3]
```diff
@@ -131,6 +159,12 @@ export default function SwapDetailsDropdown({
...
+
+  useEffect(() => {
```
Comment (cmcewen) [inline src/components/swap/SwapDetailsDropdown.tsx:165]: again can't think of a better way to do this right now but i do see we are mixing logging concerns with react state, which i think should be separate

File: src/components/swap/SwapDetailsDropdown.tsx (hunk 1 above)
Comment (lynnshaoyu) [reply src/components/swap/SwapDetailsDropdown.tsx:27]: sounds good done

File: src/components/swap/SwapDetailsDropdown.tsx (hunk 2 above)
Comment (lynnshaoyu) [reply src/components/swap/SwapDetailsDropdown.tsx:163]: yes! sorry mistakenly thought computeRealizedLPFeePercent was a custom react hook and had to be used in the component (but it's not). moved the calc to formatAnalyticsEventProperties.

File: src/components/swap/SwapDetailsDropdown.tsx (hunk 3 above)
Comment (lynnshaoyu) [reply src/components/swap/SwapDetailsDropdown.tsx:165]: hmm yeah this was my best attempt at a decent way to capture this, can ask for @zzmp's input too when he's back.

**Review by cmcewen — APPROVED** [review]
LGTM!

**Pull request merged**